web/models/campaign.ex

```diff
@@ -15,17 +15,36 @@ defmodule Jingle.Campaign do
     belongs_to :sponsor, Jingle.Sponsor
     belongs_to :podcast, Jingle.Podcast
     has_many :creatives, Jingle.Creative
-    timestamps()
+    timestamps(type: :utc_datetime)
   end
 
   @doc """
   Builds a changeset based on the `struct` and `params`.
   """
   def changeset(struct, params \\ %{}) do
     struct
-    |> cast(params, [:sponsor_id, :podcast_id, :start_date, :end_date, :due_date, :original_copy, :edited_copy, :must_say, :notes, :approved, :zone])
-    |> validate_required([:start_date, :end_date, :original_copy, :zone])
+    |> cast(utc_compatible_dates(params), [:sponsor_id, :podcast_id, :start_date, :end_date, :due_date, :original_copy, :edited_copy, :must_say, :notes, :approved, :zone])
+    |> validate_required([:start_date, :end_date, :due_date, :original_copy, :zone])
     |> assoc_constraint(:sponsor)
     |> assoc_constraint(:podcast)
   end
+
+  defp utc_compatible_dates(params) do
+    date_format = "{YYYY}-{0M}-{0D}"
+    try do
+      params = atomize_map(params)
+      %{ params | start_date: Timex.parse!(params[:start_date], date_format),
+                  end_date: Timex.parse!(params[:end_date], date_format),
+                  due_date: Timex.parse!(params[:due_date], date_format) }
+
+    rescue
```

Comment:
It's actually more idiomatic in Elixir to not throw errors, but instead match the `:ok` / `:error` in the tuple.  Something like:

```
case Timex.parse(date_str, date_format) do
  {:ok, dtim} -> dtim,
  {:error, _} -> date_str
end
```

And you may also want to convert those naive datetimes ... there's an example of something similar in [the Castle plugs](https://github.com/PRX/castle.prx.org/blob/master/lib/plugs/interval/time_to.ex).

Reply:
:+1: